include/engine/routing_algorithms/many_to_many.hpp

```diff
@@ -143,10 +143,30 @@ class ManyToManyRouting final
                     (*result_table)[source_id * number_of_targets + target_id];
                 // check if new distance is better
                 const EdgeWeight new_distance = source_distance + target_distance;
-                if (new_distance >= 0 && new_distance < current_distance)
+                if (new_distance < 0)
                 {
-                    (*result_table)[source_id * number_of_targets + target_id] =
-                        (source_distance + target_distance);
+                    for (auto edge : super::facade->GetAdjacentEdgeRange(node))
```

Comment:
We could increase the memory overhead to have a lookup. Since this should only happen at the start of a segment, I guess the overhead to compute it, is worth it to save the additional entry for every node.

Moved the lookup to Routing base.